credentials/credentials.go

```diff
@@ -126,6 +126,10 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+func (t TLSInfo) ConnState() tls.ConnectionState {
```

Comment:
On Tue, Sep 8, 2015 at 10:50 AM, Tamir Duberstein <notifications@github.com>
wrote:

> In credentials/credentials.go
> <https://github.com/grpc/grpc-go/pull/328#discussion_r38957015>:
>
> > @@ -126,6 +126,10 @@ func (t TLSInfo) AuthType() string {
> >  	return "tls"
> >  }
> >
> > +func (t TLSInfo) ConnState() tls.ConnectionState {
>
> I'm pretty sure it does prohibit mutation; it returns a copy.
>
Making it exported also makes a copy when users work on it (notice
tls.ConnectionState is a object not a pointer). Plus, making a copy does
NOT prohibit mutation because tls.ConnectionState contains a number of
pointer fields.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/328/files#r38957015>.
>